### PR TITLE
 React1-week3-Homework

### PR DESCRIPTION
**Pull request by Dima-Kutaini**

### CODE REVIEW

File: react/react1/week3/week3/src/component/TodoList.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import AddTodo from './AddTodo';
+import TodoItem from './TodoItem';
+
+let nextId = 3;
+
+const TodoList = () => {
+  const [todos, setTodos] = useState([]);
+  useEffect(() => {
+    // fetch data
+    const todosFetch = async () => {
```
Comment (LucasIversen) [inline react/react1/week3/week3/src/component/TodoList.jsx]: This probably should be wrapped in a try/catch block

File: react/react1/week3/week3/src/component/TodoList.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      console.log(data);
+      // set state when the data received
+      setTodos(data);
+    };
+
+    todosFetch();
+  }, []);
+
+  function handleAddTodo(description, deadline) {
+    setTodos([
```
Comment (LucasIversen) [inline react/react1/week3/week3/src/component/TodoList.jsx:34]: Good

File: react/react1/week3/week3/src/component/TodoList.jsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  //       id: nextId++,
+  //       deadline: deadline,
+  //     },
+  //   ]);
+  // }
+  function handleDone(id) {
+    setTodos(
+      todos.map((todo) =>
+        todo.id === id
+          ? {
```
Comment (LucasIversen) [inline react/react1/week3/week3/src/component/TodoList.jsx:57]: Nice 👌

File: react/react1/week3/week3/src/component/TodoList.jsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      />
+
+      {todos.length > 0 ? (
+        <ul>
+          {todos.map((todo) => (
+            <TodoItem
+              todo={todo}
+              deadline={todo.deadline}
+              onDone={handleDone}
+              onchange={handleChangeTodo}
```
Comment (LucasIversen) [inline react/react1/week3/week3/src/component/TodoList.jsx:94]: You've caught on to the callback pattern really well 👍

File: react/react1/week3/week3/src/component/AddTodo.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            onChange={(newDeadline) => setDealine(newDeadline)}
+            dateFormat="dd-MM-yyyy"
+            placeholder="dd-MM-yyyy"
+          />
+      </div>
+      <button
+        className="button1"
+        onClick={() => {
+          setDescription('') && setDealine('');
+          onAddTodo(description, deadline.toISOString().substring(0, 10));
```
Comment (LucasIversen) [inline react/react1/week3/week3/src/component/AddTodo.jsx:38]: Niiice 👍

File: react/react1/week3/week3/src/component/AddTodo.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      <input
+        placeholder="description "
+        value={description}
+        onChange={(e) => setDescription(e.target.value)}
+      />
+      <br />
+      <div className="deadline">
+        <lable> Deadline </lable>
+      
+        <DatePicker
```
Comment (LucasIversen) [inline react/react1/week3/week3/src/component/AddTodo.jsx:26]: I see you've found a node module to help you with the dates. Really good job

File: react/react1/week3/week3/src/component/TodoItem.jsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      <li
+        className="todo-list"
+        key={todo.id}>
+        <input
+          type="checkbox"
+          checked={todo.done}
+          onChange={() => onDone(todo.id)}
+        />
+        <p style={todo.done ? { textDecoration: 'line-through' } : {}}>
+          {content}
```
Comment (LucasIversen) [inline react/react1/week3/week3/src/component/TodoItem.jsx:45]: Good!

**Review by LucasIversen — COMMENTED** [review]
Hi Dima! Good job. The only thing I'd change is wrapping your fetch in a try/catch block to handle any errors. Other than that it's spot on 👍

File: react/react1/week3/week3/src/component/TodoItem.jsx (hunk 7 above)
Comment (Dima-Kutaini) [reply react/react1/week3/week3/src/component/TodoItem.jsx:45]: Thanks, Lucas for the review. I will update the fetch as you recommended.

**Pull request merged**